### PR TITLE
feat: Use dedicated tokio executor (aka thread pool) for DataFusion plans (DRAFT, NOT FOR MERGING)

### PR DESCRIPTION
**Pull request by alamb**
In case anyone wants to see the high level approach I plan to take, this PR shows the approach of running datafusion plans on their own `Executor` (and thus their own threadpool)

I plan on encapsulating this pattern behind a reasonable abstraction for merging, but I wanted to put up a PR with the basic approach

re https://github.com/influxdata/conductor/issues/244

### CODE REVIEW

File: query/src/exec/context.rs [hunk 1]
```diff
@@ -154,13 +154,49 @@ impl IOxExecutionContext {
...
+        // /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.4.0/src/
+        // runtime/blocking/shutdown.rs:51:21
+
+        use std::thread;
+
+        // This is definitely not cool long term, but short term just
+        // block this tokio executor while the plan is running
+        println!("Starting to run plan on separate thread...");
+
+        thread::spawn(|| {
```
Comment (alamb) [inline query/src/exec/context.rs:181]: The key is to make a new thread on which to spawn the new executor, etc

Comment (mkmik) [reply query/src/exec/context.rs:181]: here you're spawning a new thread pool for each query right, one with 4 threads each?

so if you run 10 queries in parallel, we'll end up with 40 threads to run those 10 queries?

This is a possible approach; the main pro is that there will be some fairness between concurrent queries
The main con is that we don't bound the number of threads a process will create. A minor con is that creating all those threads for each query incurs in some overheads.

Can we just create one executor upfront and then just spawn a task into it from here?

Comment (alamb) [reply query/src/exec/context.rs:181]: > here you're spawning a new thread pool for each query right, one with 4 threads each?

That is my understanding of what this PR's code does

> Can we just create one executor upfront and then just spawn a task into it from here?

@mkmik  this is a good idea -- I will give it a try

**Comment by alamb** [comment]
superceded by https://github.com/influxdata/influxdb_iox/pull/1191

**Pull request closed without merging**